### ISSUE TITLE
Selecting ingredients crashes the app

### ISSUE THREAD

**Issue by teolemon** [openfoodfacts/openfoodfacts-androidapp#2753]
**Describe the bug**
Selecting ingredients crashes the app

**To Reproduce**
1. Go to 6130743016779
2. Click on ingredients tabs
3. Select the ingredient list
4. See crash

```
s 6
2019-06-17 12:48:31.138 1092-1859/? I/AssistStructure: Flattened final assist data: 20092 bytes, containing 1 windows, 134 views
2019-06-17 12:48:31.145 1092-1092/? E/RemoteAugmentedAutofillService: Bad service name for flags 1048704: com.google.android.as/com.google.android.apps.miphone.aiai.app.AiAiAugmentedAutofillService
2019-06-17 12:48:32.332 1092-1130/? I/Process: Sending signal. PID: 20551 SIG: 9
2019-06-17 12:48:32.350 627-627/? I/Zygote: Process 20551 exited due to signal 9 (Killed)
2019-06-17 12:48:33.900 1092-1859/? D/WificondControl: Scan result ready event
2019-06-17 12:48:33.917 1092-1415/? D/WificondScannerImpl: Filtering out 12 scan results.
2019-06-17 12:48:33.974 1092-3177/? W/AppOps: Noting op not finished: uid 10013 pkg com.google.android.gms code 41 time=1560768503702 duration=0
2019-06-17 12:48:34.306 1092-3177/? W/VibratorService: Failed to perform effect (5)
2019-06-17 12:48:34.311 662-662/? E/VibratorService: Setting amplitude  to: 3596
2019-06-17 12:48:35.334 3430-3430/? E/InputEventReceiver: Exception dispatching input event.
2019-06-17 12:48:35.335 3430-3430/? D/AndroidRuntime: Shutting down VM
2019-06-17 12:48:35.336 3430-3430/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.openfoodfacts.scanner, PID: 3430
    java.lang.ArrayIndexOutOfBoundsException: length=45; index=-1
        at android.text.Layout.primaryIsTrailingPreviousAllLineOffsets(Layout.java:1125)
        at android.text.Layout.getLineHorizontals(Layout.java:1250)
        at android.text.Layout.access$000(Layout.java:51)
        at android.text.Layout$HorizontalMeasurementProvider.init(Layout.java:1654)
        at android.text.Layout$HorizontalMeasurementProvider.<init>(Layout.java:1645)
        at android.text.Layout.getOffsetForHorizontal(Layout.java:1551)
        at android.text.Layout.getOffsetForHorizontal(Layout.java:1526)
        at android.widget.TextView.getOffsetAtCoordinate(TextView.java:12695)
        at android.widget.Editor$SelectionModifierCursorController.updateWordBasedSelection(Editor.java:6121)
        at android.widget.Editor$SelectionModifierCursorController.updateSelection(Editor.java:6049)
        at android.widget.Editor$SelectionModifierCursorController.onTouchEvent(Editor.java:6019)
        at android.widget.Editor.onTouchEvent(Editor.java:1492)
        at android.widget.TextView.onTouchEvent(TextView.java:10822)
        at android.view.View.dispatchTouchEvent(View.java:13417)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
        at com.android.internal.policy.DecorView.superDispatchTouchEvent(DecorView.java:460)
        at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1849)
        at android.app.Activity.dispatchTouchEvent(Activity.java:3974)
        at android.support.v7.view.i.dispatchTouchEvent(WindowCallbackWrapper.java:68)
        at com.android.internal.policy.DecorView.dispatchTouchEvent(DecorView.java:418)
        at android.view.View.dispatchPointerEvent(View.java:13676)
        at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:5479)
        at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:5282)
        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4785)
        at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4838)
        at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4804)
        at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:4944)
        at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4812)
        at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:5001)
        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4785)
        at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4838)
        at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4804)
        at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4812)
        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4785)
        at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:7502)
2019-06-17 12:48:35.336 3430-3430/? E/AndroidRuntime:     at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:7471)
        at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:7432)
        at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:7627)
        at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:188)
        at android.view.InputEventReceiver.nativeConsumeBatchedInputEvents(Native Method)
        at android.view.InputEventReceiver.consumeBatchedInputEvents(InputEventReceiver.java:178)
        at android.view.ViewRootImpl.doConsumeBatchedInput(ViewRootImpl.java:7578)
        at android.view.ViewRootImpl$ConsumeBatchedInputRunnable.run(ViewRootImpl.java:7651)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:966)
        at android.view.Choreographer.doCallbacks(Choreographer.java:790)
        at android.view.Choreographer.doFrame(Choreographer.java:718)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:951)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7319)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:934)
2019-06-17 12:48:35.340 1092-1442/? W/ActivityTaskManager:   Force finishing activity org.openfoodfacts.scanner/openfoodfacts.github.scrachx.openfood.views.AddProductActivity
2019-06-17 12:48:35.340 1092-3606/? I/DropBoxManagerService: add tag=data_app_crash isTagEnabled=true flags=0x2
2019-06-17 12:48:35.347 1092-3606/? W/DropBoxManagerService: Dropping: data_app_crash (10 > 0 bytes)
2019-06-17 12:48:35.349 3430-3430/? I/Process: Sending signal. PID: 3430 SIG: 9
2019-06-17 12:48:35.356 1092-1130/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
2019-06-17 12:48:35.358 1092-1130/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
2019-06-17 12:48:35.385 1092-1382/? W/InputDispatcher: channel '3dd8915 org.openfoodfacts.scanner/openfoodfacts.github.scrachx.openfood.views.MainActivity (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
2019-06-17 12:48:35.385 1092-1382/? E/InputDispatcher: channel '3dd8915 org.openfoodfacts.scanner/openfoodfacts.github.scrachx.openfood.views.MainActivity (server)' ~ Channel is unrecoverably broken and will be disposed!
2019-06-17 12:48:35.387 1092-1842/? I/WindowManager: WIN DEATH: Window{3dd8915 u0 org.openfoodfacts.scanner/openfoodfacts.github.scrachx.openfood.views.MainActivity}
2019-06-17 12:48:35.387 1092-1842/? W/InputDispatcher: Attempted to unregister already unregistered input channel '3dd8915 org.openfoodfacts.scanner/openfoodfacts.github.scrachx.openfood.views.MainActivity (server)'
2019-06-17 12:48:35.387 1092-1495/? I/ActivityManager: Process org.openfoodfacts.scanner (pid 3430) has died: fore TOP 
2019-06-17 12:48:35.390 1092-3178/? I/WindowManager: WIN DEATH: Window{7cc1ba8 u0 org.openfoodfacts.scanner/openfoodfacts.github.scrachx.openfood.views.AddProductActivity}
2019-06-17 12:48:35.393 1092-6221/? I/WindowManager: WIN DEATH: Window{a0f2ead u0 org.openfoodfacts.scanner/openfoodfacts.github.scrachx.openfood.views.product.ProductActivity}
2019-06-17 12:48:35.396 1092-1118/? I/WindowManager: WIN DEATH: Window{403eedf u0 org.openfoodfacts.scanner/openfoodfacts.github.scrachx.openfood.views.HistoryScanActivity}
2019-06-17 12:48:35.416 1092-1125/? E/system_server: Invalid ID 0x00000000.
2019-06-17 12:48:35.417 1092-1124/? W/ActivityManager: setHasOverlayUi called on unknown pid: 3430
2019-06-17 12:48:35.431 1092-1132/? I/libprocessgroup: Successfully killed process cgroup uid 10446 pid 3430 in 44ms
2019-06-17 12:48:35.447 626-626/? D/Zygote: Forked child process 3613
2019-06-17 12:48:35.447 626-626/? I/Zygote: Process 3430 exited due to signal 9 (Killed)
2019-06-17 12:48:35.449 1092-1131/? I/ActivityManager: Start proc 3613:org.openfoodfacts.scanner/u0a446 for activity {org.openfoodfacts.scanner/openfoodfacts.github.scrachx.openfood.views.product.ProductActivity}
2019-06-17 12:48:35.472 3613-3613/? E/odfacts.scanne: Not starting debugger since process cannot load the jdwp agent.
2019-06-17 12:48:35.508 3613-3613/? I/odfacts.scanne: The ClassLoaderContext is a special shared library.
``` 
**Smartphone (please complete the following information):**
 - Device: Google Pixel 1
 - OS: Android 9
 - Version 317

**Comment by deniger** [comment]
will bar difficult to solve as only Android classes are related to this issue...

**Comment by deniger** [comment]
Fixed by #2751

**Comment by teolemon** [comment]
thx. Not sure why it's not auto-closing. 
Closing
